src/applicationManager.ts

```diff
@@ -528,3 +546,228 @@ async function getGraphAPIRoles(token: string, graphApiPermissions: string[]) {
   }
   return { appRoleIds, graphAPIObjectId };
 }
+
+/**
+ * Updates an Azure Entra Application, specifically with a collection of new App Roles
+ * @param applicationId The object ID of application registration
+ * @param appRoles A collection (array) of AppRole custom objects to update the application with
+ */
+export async function updateApplicationAppRoles({
+  token,
+  applicationId,
+  appRoles,
+}: {
+  token: string;
+  applicationId: string;
+  appRoles: Array<AppRole>;
+}): Promise<void> {
+  let body = {
+    appRoles: appRoles,
+  };
+  const result = await fetch(
+    `https://graph.microsoft.com/v1.0/applications/${applicationId}`,
+    {
+      method: "PATCH",
+      headers: {
+        Authorization: `Bearer ${token}`,
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify(body),
+    },
+  );
+  await errorHandler("updating application config", result);
+}
+
+/**
+ * Create formatted AppRole object which is expected by graph api
+ * @param id Must be unique compared to other App Roles on this application
+ * @param value Must be unique compared to other App Roles on this application
+ * @return AppRole Returns a promise of structured AppRole type object
+ */
+async function generateStructuredAppRole(
+  displayName: string,
+  description: string,
+  id: string,
+  value: string,
+): Promise<AppRole> {
```

Comment:
this doesn't return a promise
```suggestion
): AppRole {
```